src/mango/src/mango_native_proc.erl

```diff
@@ -257,15 +257,10 @@ add_default_text_field([_ | Rest], Acc) ->
 
 
 %% index of all field names
-get_field_names([], FAcc) ->
-    FAcc;
-get_field_names([{Name, _Type, _Value} | Rest], FAcc) ->
-    case lists:member([<<"$fieldnames">>, Name, []], FAcc) of
-        true ->
-            get_field_names(Rest, FAcc);
-        false ->
-            get_field_names(Rest, [[<<"$fieldnames">>, Name, []] | FAcc])
-    end.
+get_field_names(Fields) ->
+    GBFieldSet = gb_sets:from_list(Fields),
```

Comment:
`Fields` is a list of size 3 tuples of following structure `{FieldName, Type, Value}`.  We would want to ignore Value. Something like this would do it.
```
Set = lists:foldl(
    fun({Name, _, _}, Set) -> 
         gb_sets:add([<<"$fieldnames">>, Name, []], Set) 
    end, gb_sets:new(), Fields),
gb_sets:to_list(Set).
```